src/include/duckdb/common/arrow/appender/list_view_data.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "duckdb/common/arrow/appender/append_data.hpp"
+
+namespace duckdb {
+
+template <class BUFTYPE = int64_t>
+struct ArrowListViewData {
+public:
+	static void Initialize(ArrowAppendData &result, const LogicalType &type, idx_t capacity) {
+		auto &child_type = ListType::GetChildType(type);
+		result.main_buffer.reserve(capacity * sizeof(BUFTYPE));
+		result.aux_buffer.reserve(capacity * sizeof(BUFTYPE));
+
+		auto child_buffer = ArrowAppender::InitializeChild(child_type, capacity, result.options);
+		result.child_data.push_back(std::move(child_buffer));
+	}
+
+	static void Append(ArrowAppendData &append_data, Vector &input, idx_t from, idx_t to, idx_t input_size) {
+		UnifiedVectorFormat format;
+		input.ToUnifiedFormat(input_size, format);
+		idx_t size = to - from;
+		vector<sel_t> child_indices;
+		AppendValidity(append_data, format, from, to);
+		AppendOffsets(append_data, format, from, to, child_indices);
+
+		// append the child vector of the list
+		SelectionVector child_sel(child_indices.data());
+		auto &child = ListVector::GetEntry(input);
+		auto child_size = child_indices.size();
+		Vector child_copy(child.GetType());
+		child_copy.Slice(child, child_sel, child_size);
+		append_data.child_data[0]->append_vector(*append_data.child_data[0], child_copy, 0, child_size, child_size);
+		append_data.row_count += size;
+	}
+
+	static void Finalize(ArrowAppendData &append_data, const LogicalType &type, ArrowArray *result) {
+		result->n_buffers = 3;
+		result->buffers[1] = append_data.main_buffer.data();
+		result->buffers[2] = append_data.aux_buffer.data();
+
+		auto &child_type = ListType::GetChildType(type);
+		ArrowAppender::AddChildren(append_data, 1);
+		result->children = append_data.child_pointers.data();
+		result->n_children = 1;
+		append_data.child_arrays[0] = *ArrowAppender::FinalizeChild(child_type, std::move(append_data.child_data[0]));
+	}
+
+public:
+	static void AppendOffsets(ArrowAppendData &append_data, UnifiedVectorFormat &format, idx_t from, idx_t to,
```

Comment:
This name was a bit confusing, we are actually filling up both offset and size buffers right, maybe we should also add a comment with the actual expected layout as described in [here](https://arrow.apache.org/docs/format/Columnar.html)